azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -101,6 +97,7 @@ def __init__(
         websockets=False,
         cipher=None,
         proxy_options=None,
+        keep_alive=None,
```

Comment:
>keep_alive=None, [](start = 8, length = 16)

As you have this written, the connection will fail if the user doesn't provide a keep_alive value, so it's not _really_ optional.  (please don't make it required).

Reply:
the value is defaulted at the config level , so it will never be None rather it will be always 180 secs...

---
In reply to: [462526084](https://github.com/Azure/azure-iot-sdk-python/pull/635#discussion_r462526084) [](ancestors = 462526084)